reviewrot/gitlabstack.py

```diff
@@ -0,0 +1,170 @@
+import os
+import logging
+import gitlab
+import datetime
+from dateutil.relativedelta import relativedelta
+from basereview import BaseService, BaseReview
+from gitlab.exceptions import GitlabGetError
+
+log = logging.getLogger(__name__)
+
+
+class GitlabService(BaseService):
+    """
+    This class represents Gitlab. The reference can be found here:
+     https://docs.gitlab.com/ee/api/
+    """
+    def request_reviews(self, user_name, repo_name=None, state_=None,
+                        value=None, duration=None, token=None, host=None):
+        """
+        Creates a gitlab object.
+        Requests merge requests for specified username and repo name.
+        If repo name is not provided then requests merge requests
+        for all repos for specified username/organization.
+
+        Args:
+            user_name (str): Gitlab namespace
+            repo_name (str): Gitlab project name for specified
+                          namespace
+            state_ (str): The state for pull requests, e.g, older
+                        or newer
+            value (int): The value in terms of duration for requests
+                         to be older or newer than
+            duration (str): The duration in terms of period(year, month,
+                            hour, minute) for requests to be older or
+                            newer than
+            token (str): Gitlab token for authentication
+            host (str): Gitlab host name for authentication
+
+        Returns:
+            response (list): Returns list of list of pull requests for
+                             specified user(group) name and projectname or all
+                             projectname for given groupname
+        """
+        gl = gitlab.Gitlab(host, token)
+        gl.auth()
+        log.debug('Gitlab instance created: %s', gl)
+        response = []
+        # if Repository name is explicitely provided
+        if repo_name is not None:
+            try:
+                # get project object for given repo_name(project name)
+                project = gl.projects.get(os.path.join(user_name, repo_name))
+            except GitlabGetError:
+                log.exception('Project %s not found for user %s',
+                              repo_name, user_name)
+                raise Exception('Project %s not found for user %s'
+                                % (repo_name, user_name))
+            # get merge requests for specified username and project name
+            res = self.get_reviews(uname=user_name, project=project,
```

Comment:
You're not doing anything with `res` after this. Should it be appended to `response`?